web.browser/src/components/NewPost/index.js

```diff
@@ -0,0 +1,203 @@
+import React, { PropTypes } from 'react';
+import { Card, CardText } from 'material-ui/Card';
+import Paper from 'material-ui/Paper';
+import { Toolbar, ToolbarTitle } from 'material-ui/Toolbar';
+import TextField from 'material-ui/TextField';
+import SelectField from 'material-ui/SelectField';
+import RaisedButton from 'material-ui/RaisedButton';
+import Gandalf from 'gandalf-validator';
+import styles from './styles.css';
+
+class NewPost extends Gandalf {
+  constructor({ selectControlItems }) {
+    const fields = {
+      title: {
+        component: TextField,
+        validators: ['required'],
+        errorPropName: 'errorText',
+        props: {
+          hintText: 'Title',
+          floatingLabelText: 'Title',
+          style: {
+            width: '100%',
+          },
+        },
+      },
+      description: {
+        component: TextField,
+        validators: ['required'],
+        errorPropName: 'errorText',
+        props: {
+          hintText: 'Description',
+          floatingLabelText: 'Description',
+          style: {
+            width: '100%',
+          },
+        },
+      },
+      /* category: {
+        component: SelectField,
+        // validators: ['required'],
+        errorPropName: 'errorText',
+        props: {
+          hintText: 'Select a category',
+          floatingLabelText: 'Category',
+          style: {
+            width: '100%',
+          },
+        },
+        selectFieldValue: null,
+      },*/
+      link: {
+        component: TextField,
+        validators: ['required'],
+        errorPropName: 'errorText',
+        props: {
+          hintText: 'Link',
+          floatingLabelText: 'Link',
+          style: {
+            width: '100%',
+          },
+        },
+      },
+      tag: {
+        component: TextField,
+        validators: ['required'],
+        errorPropName: 'errorText',
+        props: {
+          hintText: 'Tag',
+          floatingLabelText: 'Tag',
+          style: {
+            width: '100%',
+          },
+        },
+      },
+    };
+
+    super(fields, selectControlItems);
+  }
+
+  handleSelectChange = (
+    (event, index, selectFieldValue) => (this.setState({ selectFieldValue }))
+  );
```

Comment:
Wrapping this function in brackets doesn't do anything for you here. This should be:

```js
handleSelectChange = (event, index, selectFieldValue) => (this.setState({ selectFieldValue });
```